core/templates/domain/story/editable-story-backend-api.service.ts

```diff
@@ -52,125 +65,123 @@ angular.module('oppia').factory('EditableStoryBackendApiService', [
+  }
+    private _updateStory = function(
+        storyId: string, storyVersion: string,
+        commitMessage: string,
+        changeList: string[],
+        successCallback: (value?: Object | PromiseLike<Object>) =>void,
+        errorCallback: (reason?: Object | PromiseLike<Object>) => void): void {
+      const editableStoryDataUrl = this.urlInterpolation.interpolateUrl(
+        StoryDomainConstants.EDITABLE_STORY_DATA_URL_TEMPLATE, {
           story_id: storyId
         });
-
       var putData = {
         version: storyVersion,
         commit_message: commitMessage,
         change_dicts: changeList
       };
-      $http.put(editableStoryDataUrl, putData).then(function(response) {
-        // The returned data is an updated story dict.
-        var story = angular.copy(response.data.story);
-
-        if (successCallback) {
-          successCallback(story);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
-      });
+      this.http.put(editableStoryDataUrl, putData).toPromise().then(
+        response => successCallback(response.story),
+        errorResponse => errorCallback(errorResponse.error)
+      );
     };
-
-    var _changeStoryPublicationStatus = function(
-        storyId, newStoryStatusIsPublic, successCallback, errorCallback) {
-      var storyPublishUrl = UrlInterpolationService.interpolateUrl(
-        STORY_PUBLISH_URL_TEMPLATE, {
+    private _changeStoryPublicationStatus = function(
+        storyId: string,
+        newStoryStatusIsPublic: boolean,
+        successCallback: (value?: Object | PromiseLike<Object>) =>void,
+        errorCallback: (reason?: Object | PromiseLike<Object>) => void): void {
+      const storyPublishUrl = this.urlInterpolation.interpolateUrl(
+        StoryDomainConstants.STORY_PUBLISH_URL_TEMPLATE, {
           story_id: storyId
         });
-
       var putData = {
         new_story_status_is_public: newStoryStatusIsPublic
       };
-      $http.put(storyPublishUrl, putData).then(function(response) {
-        if (successCallback) {
-          successCallback();
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
-      });
+      this.http.put(storyPublishUrl, putData).toPromise().then(
+        response => successCallback(response.status),
+        errorResponse => errorCallback(errorResponse.error));
     };
 
-    var _validateExplorations = function(
-        storyId, expIds, successCallback, errorCallback) {
-      var validateExplorationsUrl = UrlInterpolationService.interpolateUrl(
-        VALIDATE_EXPLORATIONS_URL_TEMPLATE, {
+    private _validateExplorations = function(
+        storyId: string,
+        expIds: string[],
+        successCallback: (value?: Object | PromiseLike<Object>) =>void,
+        errorCallback: (reason?: Object | PromiseLike<Object>) => void): void {
```

Comment:
Exact type here and elsewhere (wherever `Object` is used as a type).

Reply:
@nishantwrp yes done here and everywhere else